src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -445,7 +446,58 @@ protected static void ShiftFieldsFromRootToInstallerLevel(InstallerManifest inst
                     property.SetValue(installerManifest, null);
                 }
             }
-        }
+        }
+
+        /// <summary>
+        /// Shifts common installer fields from installer level to manifest root.
+        /// </summary>
+        /// <param name="installerManifest">Wrapper object containing the installer manifest object models.</param>
+        protected static void ShiftInstallerFieldsToRootLevel(InstallerManifest installerManifest)
+        {
+            var rootProperties = installerManifest.GetType().GetProperties();
+            var installerProperties = installerManifest.Installers.First().GetType().GetProperties();
+
+            // Get common properties between root and installer level
+            var commonProperties = rootProperties.Where(rp => installerProperties.Any(ip => ip.Name == rp.Name));
+
+            foreach (var property in commonProperties)
+            {
+                var firstInstallerValue = installerManifest.Installers.First().GetType().GetProperty(property.Name).GetValue(installerManifest.Installers.First());
+                if (firstInstallerValue != null)
+                {
+                    // Check if all installers have the same value
+                    bool allInstallersHaveSameValue = installerManifest.Installers.All(i =>
+                    {
+                        var propertyValue = i.GetType().GetProperty(property.Name).GetValue(i);
+                        return propertyValue != null &&
+                        propertyValue.Equals(firstInstallerValue);
+                    });
+
+                    // If value is false, it can be becuase we don't have .Equals() override implemented for the type of the property.

```

Comment:
```suggestion
                    // If value is false, it can be because we don't have .Equals() override implemented for the type of the property.

```